src/wallet/wallet.cpp

```diff
@@ -362,6 +406,42 @@ bool CWallet::AddCryptedSaplingSpendingKey(const libzcash::SaplingExtendedFullVi
     return false;
 }
 
+UnifiedSpendingKey CWallet::GenerateNewUnifiedSpendingKey() {
+    AssertLockHeld(cs_wallet);
+
+    CHDChain& hdChain = mnemonicHDChain.value();
+    while (true) {
+        auto usk = GenerateUnifiedSpendingKeyForAccount(hdChain.GetAccountCounter());
+        hdChain.IncrementAccountCounter();
+
+        if (usk.has_value()) {
+            // Update the persisted chain information
+            if (fFileBacked && !CWalletDB(strWalletFile).WriteMnemonicHDChain(hdChain)) {
+                throw std::runtime_error(
+                        "CWallet::GenerateNewUnifiedSpendingKey(): Writing HD chain model failed");
+            }
+
+            return usk.value();
+        }
+    }
+}
+
+std::optional<libzcash::UnifiedSpendingKey> CWallet::GenerateUnifiedSpendingKeyForAccount(uint32_t accountId) {
+    auto seed = GetMnemonicSeed();
+    if (!seed.has_value()) {
+        throw std::runtime_error(std::string(__func__) + ": Wallet has no mnemonic HD seed. Please upgrade this wallet.");
```

Comment:
```suggestion
        throw std::runtime_error(std::string(__func__) + ": Wallet has no mnemonic HD seed.");
```
Since the inclusion of a mnemonic seed is going to be automatic now, there's not an upgrade that the user can perform so I think we just treat this as a fatal error requiring developer intervention.